extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java

```diff
@@ -546,6 +547,19 @@ public void handle(RoutingContext event) {
             // Add body handler and cors handler
             var mr = managementRouter.getValue();
 
+            if (hotReplacementHandler != null) {
+                //recorders are always executed in the current CL
+                ClassLoader currentCl = Thread.currentThread().getContextClassLoader();
+                mr.route().order(RouteConstants.ROUTE_ORDER_HOT_REPLACEMENT)
+                        .handler(new Handler<RoutingContext>() {
+                            @Override
+                            public void handle(RoutingContext event) {
+                                Thread.currentThread().setContextClassLoader(currentCl);
+                                hotReplacementHandler.handle(event);
```

Comment:
I'm surprised we don't restore the previous CL, here?

Reply:
I can only assume that it's probably so that all subsequent handlers would run with this `currentCl` and it could be  restored in the very last handler, but there's probably not much benefit in that?